### ISSUE TITLE
Database engine support

### ISSUE THREAD

**Issue by pat-s** [vimc/vaultr#35]
Vault allows generating (rotated) DB credentials with a short TTL.

Support for this in {vaultr} would be great, eventually with an integration into {odbc}.
E.g. a call to {odbc} could have a flag indicating that the auth credentials should be generated via {vaultr} in the background, eventually with an option which policy to use (if multiple ones are present). This is surely something for christmas in a few years probably, but would be really nice to aim for I guess?

**Comment by richfitz** [comment]
yes, this would be nice - happy for someone to scope it out further or to look at a PR. It's unlikely that I'll get to it very soon though
